src/components/Header.js

```diff
@@ -34,16 +48,34 @@ const Header = () => {
                </Link>
             </Drawer>
          )}
-         <Box>Logo</Box>
+         <ReactLink to='/'>
+            <Heading as='h1' size='md'>
+               uTeam
+            </Heading>
+         </ReactLink>
 
-         <Flex alignItems='center' justifyContent='space-between'>
+         <Flex
+            alignItems='center'
+            justifyContent='space-between'
+            fontWeight='semibold'
+         >
             {isLargerThan768 ? (
                <Flex alignItems='center' justifyContent='space-between'>
-                  <Link as={ReactLink} to='login' mr='1rem'>
-                     Login
+                  <Link as={ReactLink} to={isLoggedIn ? '/profile' : '/login'}>
+                     {isLoggedIn && userProfile
+                        ? `${userProfile.name}`
+                        : 'Login'}
```

Comment:
Prefer conditionally rendering whole components instead of props / children

```suggestion
{isLoggedIn ? (
  <Link as={ReactLink} to="/profile">
      {userProfile?.name}
  </Link>
) : (
  <Link as={ReactLink} to="/login">
      Login
  </Link>
)
```